asreview/settings.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC
+import os
+from asreview.utils import config_from_file, _safe_dict_update
+
+
+class ASReviewSettings(ABC):
+    def __init__(self, model, n_instances, n_queries, n_prior_included,
+                 n_prior_excluded, query_strategy,
+                 balance_strategy, mode, data_fp):
+        data_name = os.path.basename(data_fp)
+        self._settings = {
```

Comment:
Unnest?

Reply:
I'm not sure what you mean?

Reply:
There is no need to wrap the settings in an additional attribute named `_settings`. Please put them on class level. 

I'm not so sure on the type checking as well. Can you clarify this?

Reply:
It seems much easier to loop over elements of a dictionary than those of class elements though?

Reply:
The configuration file doesn't have any information about type, so instead of guessing, I explicitly convert it using these stored types.